From 7ee78ee9c4aa640bfc0ba68d8d2ad0a63341a9f5 Mon Sep 17 00:00:00 2001
From: Nicholas Guriev <nicholas@guriev.su>
Date: Wed, 20 Apr 2022 12:34:08 +0300
Subject: [PATCH] Seek for Wayland protocols definitions in a system-wide
 package first

And if not found, fallback to vendored XMLs.
---
 waylandshells/xdg-shell/CMakeLists.txt | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/waylandshells/xdg-shell/CMakeLists.txt b/waylandshells/xdg-shell/CMakeLists.txt
index a4e3674..3565904 100644
--- a/Telegram/lib_waylandshells/waylandshells/xdg-shell/CMakeLists.txt
+++ b/Telegram/lib_waylandshells/waylandshells/xdg-shell/CMakeLists.txt
@@ -8,6 +8,7 @@ if (NOT DESKTOP_APP_QT6)
     find_package(ECM REQUIRED)
     set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
     find_package(QtWaylandScanner REQUIRED)
+    find_package(WaylandProtocols)
 endif()
 
 add_library(lib_waylandshells_xdg_shell STATIC)
@@ -17,26 +18,32 @@ init_target(lib_waylandshells_xdg_shell)
 get_filename_component(src_loc . REALPATH)
 set_target_properties(lib_waylandshells_xdg_shell PROPERTIES AUTOMOC ON)
 
+if (WaylandProtocols_FOUND)
+    set(wayland_protocols_loc ${WaylandProtocols_DATADIR})
+else()
+    set(wayland_protocols_loc ${third_party_loc}/wayland-protocols)
+endif()
+
 if (DESKTOP_APP_QT6)
     qt6_generate_wayland_protocol_client_sources(lib_waylandshells_xdg_shell
     FILES
-        ${third_party_loc}/wayland-protocols/stable/xdg-shell/xdg-shell.xml
-        ${third_party_loc}/wayland-protocols/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml
-        ${third_party_loc}/wayland-protocols/staging/xdg-activation/xdg-activation-v1.xml
+        ${wayland_protocols_loc}/stable/xdg-shell/xdg-shell.xml
+        ${wayland_protocols_loc}/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml
+        ${wayland_protocols_loc}/staging/xdg-activation/xdg-activation-v1.xml
     )
 else()
     ecm_add_qtwayland_client_protocol(xdg_shell_protocol_srcs
-        PROTOCOL ${third_party_loc}/wayland-protocols/stable/xdg-shell/xdg-shell.xml
+        PROTOCOL ${wayland_protocols_loc}/stable/xdg-shell/xdg-shell.xml
         BASENAME xdg-shell
     )
 
     ecm_add_qtwayland_client_protocol(xdg_decoration_protocol_srcs
-        PROTOCOL ${third_party_loc}/wayland-protocols/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml
+        PROTOCOL ${wayland_protocols_loc}/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml
         BASENAME xdg-decoration-unstable-v1
     )
 
     ecm_add_qtwayland_client_protocol(xdg_activation_protocol_srcs
-        PROTOCOL ${third_party_loc}/wayland-protocols/staging/xdg-activation/xdg-activation-v1.xml
+        PROTOCOL ${wayland_protocols_loc}/staging/xdg-activation/xdg-activation-v1.xml
         BASENAME xdg-activation-v1
     )
 endif()
